episodes/import-and-visualise-osm-data.Rmd

```diff
@@ -0,0 +1,226 @@
+
+Let's say we want to download data from OpenStreetMap and we know there is a package for it named `osmdata`, but we don't know which function to use and what arguments are needed. Where should we start?
+
+> Let's check the documentation [online](https://docs.ropensci.org/osmdata/):
+
+![The OSMdata Documentation page](https://raw.githubusercontent.com/ClementineCttn/r-geospatial-urban/blob/main/episodes/fig/osmdata.png){width=80%}
+
+It appears that there is a function to extract features, using the Overpass API. This function's name is `opq` (for OverPassQuery) which, in combination with `add_osm_feature`, seems to do the job. However it might not be crystal clear how  to apply it to our case. Let's click on the function name to know more.
+
+![The Overpass Query Documentation page](https://raw.githubusercontent.com/ClementineCttn/r-geospatial-urban/blob/main/episodes/fig/opq.png){width=80%}
+
+
+
+On this page we can read about the arguments needed for each function: a bounding box for `opq()` and some `key` and `value` for `add_osm_feature()`. Thanks to the examples provided, we can assume that these keys and values correspond to different levels of tags from the OSM classification. In our case, we will keep it at the first level of classification, with "buildings" as `key`, and no value. We also see from the examples that another function is needed when working with the `sf` package: `osmdata_sf()`. This ensures that the type of object is suited for `sf`. With these tips and examples, we can write our feature extraction function as follows:
+
+
+```{r}
+x <- opq(bbox = bb) %>%
+   add_osm_feature(key = 'building') %>%
+    osmdata_sf()
+```
+
+
+
+### Structure of objects
+
+What is this x object made of? It is a table of all the buildings contained in the bounding box, which gives us their OSM id, their geometry and a range of attributes, such as their name, building material, building date, etc. The completion level of this table depends on user contributions and open resources (here for instance: BAG, different in other countries).
+
+
+
+```{r}
+head(x$osm_polygons)
+```
+
+
+
+## Mapping 
+
+
+Let's map the building age of post-1900 Brielle buildings.
+
+
+### Projections
+
+First, we are going to select the polygons and reproject them with the Amersfoort/RD New projection, suited for maps centred on the Netherlands. This code for this projection is: 28992.
+
+```{r}
+buildings <- x$osm_polygons %>%
+  st_transform(.,crs=28992)
+```
+
+:::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::: instructor
+
+If this does work for some reason, the `buildings` can be found in the data folder: "episodes/data/dataBrielle.shp".
+
+::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
+
+
+
+### Visualisation
+
+Then we create a variable which a threshold at 1900. Every date prior to 1900 will be recoded 1900, so that buildings older than 1900 will be represented with the same shade.
```

Comment:
```suggestion
Then we create a variable using a threshold at 1900. Every date before 1900 will be recoded as `1900`, so that buildings older than 1900 will be represented with the same shade.
```